receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -388,23 +389,115 @@ func addNumberDatapoints(datapoints pmetric.NumberDataPointSlice, ls labels.Labe
 		dp.SetDoubleValue(sample.Value)
 
 		attributes := dp.Attributes()
-		for _, l := range ls {
-			if l.Name == "instance" || l.Name == "job" || // Become resource attributes
-				l.Name == labels.MetricName || // Becomes metric name
-				l.Name == "otel_scope_name" || l.Name == "otel_scope_version" { // Becomes scope name and version
-				continue
-			}
-			attributes.PutStr(l.Name, l.Value)
-		}
+		extractAttributes(ls).CopyTo(attributes)
 	}
 }
 
 func addSummaryDatapoints(_ pmetric.SummaryDataPointSlice, _ labels.Labels, _ writev2.TimeSeries) {
 	// TODO: Implement this function
 }
 
-func addHistogramDatapoints(_ pmetric.HistogramDataPointSlice, _ labels.Labels, _ writev2.TimeSeries) {
-	// TODO: Implement this function
+func addExponentialHistogramDatapoints(datapoints pmetric.ExponentialHistogramDataPointSlice, ls labels.Labels, ts writev2.TimeSeries) {
+	for _, histogram := range ts.Histograms {
+		dp := datapoints.AppendEmpty()
```

Comment:
Should we have this stale nan check? https://github.com/perebaj/opentelemetry-collector-contrib/blob/e653f32b3b15110d75f9cd92fd0370024b4d8732/receiver/prometheusreceiver/internal/metricfamily.go#L171C6-L171C22

Reply:
Could you explain how staleness related to PRW?

Reply:
@nicolastakashi following the krajo work, I think that yes, but I don't know why this is important... Trying to figure out this with the specialists 😇

Reply:
Staleness markers are not something specific to Histograms, right? Would it make sense if we tackle them in a separate PR? And with all metric types?

I'm a bit concerned that we're trying to build something perfect here and never deliver. I'd love to see us tackling our tasks in small deliverables, especially since this component has not even been released yet